src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -328,6 +334,49 @@ protected ulong ClearRegister
             set {  *(ulong*)(_registerViewPointer->GPCLR) = value; }
         }
 
+        /// <summary>
+        /// Returns the pperipheral base address of the raspberry pi based on the ranges set within the device tree.
+        /// The return value is 32bit even on 64 bit operating systems (debian / ubuntu tested) but may change in the future
+        /// 
+        /// Based on file at https://github.com/RPi-Distro/raspi-gpio/blob/master/raspi-gpio.c
+        /// </summary>
+        /// <returns>This reurns the peripheral base address as a 32 bit address.</returns>
+        private uint GetPeripheralBaseAddress()
```

Comment:
Given that the driver is really only supose to work with the BCM2835, isn't this address always the same?

Reply:
Well I believe that the BCM2835 was used in the Pi 1, Pi 2 has the 2836, Pi 3 2837 and the Pi4 2711. For software sake the system just reports BCM2835 but they really are different models with IO mapped to different addresses. So to determine where to write the IO data for different models we need to read the info from the device tree. /dev/gpiomem does this for use but for /dev/mem we have to work it out.

Reply:
@joperezr @krwq 
I am intending to investigate using a function in libbcm_host to determine the Peripheral Base Address via interrop. This is described in the PI documentation as the correct way to go. If it works then the conversations about the home grown method will be unnecessary I guess.

Reply:
> update: It looks like the userland library libbcm_host is not installed by default in Debian and ubuntu. Given this perhaps we should continue with the GetPeripheralBaseAddress method.

I see, yeah or another think we could try is to PInvoke and see if the method is there and if we fail then we fallback to this method

Reply:
@joperezr @krwq The latest PR update calls libbcm_host for the base address and fails over to the GetPeripheralBaseAddress method if the library isn't installed.

I have also done more commenting around this method.

The PR update seems to file on timeout for helix tests